utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -117,55 +119,64 @@ impl TrieType for Small {
 /// For more information:
 /// - [ICU Site design doc](http://site.icu-project.org/design/struct/utrie)
 /// - [ICU User Guide section on Properties lookup](https://unicode-org.github.io/icu/userguide/strings/properties.html#lookup)
-pub struct CodePointTrie<'trie, W: ValueWidth, T: TrieType> {
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct CodePointTrie<'trie, W: ValueWidth>
+where <<W as AsULE>::ULE as ULE>::Error: Display
```

Comment:
One of the compiler errors was that, as I understood it, the default implementation of Serde provided by the derive required that the Error type implement `std::fmt::Display`, but instead, the added trait bound uses `core::fmt::Display`. Is that ok?

Reply:
We should avoid putting trait bounds on the struct.  Trait bounds belong on impls.  Serde complaining isn't a good reason to put a trait bound on a struct.  Manish shared a really good article on the subject last week: https://stackoverflow.com/questions/49229332/should-trait-bounds-be-duplicated-in-struct-and-impl/66369912#66369912